crypto.go

```diff
@@ -167,10 +217,29 @@ func (s ecdhScheme) GenerateKeyPair(rand io.Reader) (KEMPrivateKey, KEMPublicKey
 }
 
 func (s ecdhScheme) Marshal(pk KEMPublicKey) []byte {
+	if pk == nil {
+		return nil
+	}
 	raw := pk.(*ecdhPublicKey)
 	return elliptic.Marshal(raw.curve, raw.x, raw.y)
 }
 
+func (s ecdhScheme) MarshalPrivate(sk KEMPrivateKey) []byte {
+	if sk == nil {
+		return nil
+	}
+
+	raw := sk.(*ecdhPrivateKey)
+	encodedPoint := elliptic.Marshal(raw.curve, raw.x, raw.y)
+
+	bundle := make([]byte, 2)
```

Comment:
No need, technically, to encode the point here, and if you get rid of that, you don't need the length.  That is, you could just `return raw.d.Bytes()`.

Reply:
Good point -- we can just recompute `(x,y)` from `d` when unmarshalling.